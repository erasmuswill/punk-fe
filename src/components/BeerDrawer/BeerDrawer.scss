.beer-drawer {
  .ant-statistic-title {
    color: #00000065;
  }

  span.ant-descriptions-item-label {
    color: #00000065;
  }

  .ant-descriptions-header {
    margin-bottom: 0px;
    margin-top: 24px;
  }
}
