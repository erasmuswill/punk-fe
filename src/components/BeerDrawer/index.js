import {
  Descriptions,
  Drawer,
  Tooltip,
  Row,
  Col,
  Button,
  Statistic,
  Table,
} from "antd";
import Title from "antd/lib/typography/Title";
import { useState } from "react";
import { useBeer } from "../../api";

function BeerDrawer({ id, close }) {
  const [showIngredients, setShowIngredients] = useState(false);
  const [showMethod, setShowMethod] = useState(false);
  const { beer } = useBeer(id);
  if (!beer) return null;
  const {
    name,
    tagline,
    first_brewed,
    abv,
    ibu,
    target_fg,
    target_og,
    ebc,
    srm,
    ph,
    attenuation_level,
    volume,
    boil_volume,
    method: { mash_temp, twist, fermentation },
    ingredients: { hops, malt, yeast },
    food_pairing,
    brewers_tips,
  } = beer;
  return (
    <Drawer
      title={name}
      width={520}
      onClose={close}
      visible={!!id && id !== 0}
      footer={
        <Row justify="space-around">
          <Col>
            <Button onClick={() => setShowIngredients(true)}>
              Show Ingredients
            </Button>
          </Col>
          <Col>
            <Button onClick={() => setShowMethod(true)}>Show Method</Button>
          </Col>
        </Row>
      }
    >
      <Title level={4}>{tagline}</Title>
      <Row justify="space-between">
        <Col>
          <Statistic title="ABV" value={abv} />
        </Col>
        <Col>
          <Statistic title="IBU" value={ibu} />
        </Col>
        <Col>
          <Statistic title="OG" value={target_og} />
        </Col>
        <Col>
          <Statistic title="First Brewed" value={first_brewed} />
        </Col>
      </Row>
      <Descriptions title="Basics">
        <Descriptions.Item label="Volume">
          <Tooltip title="Amount of wort at the end of the brewing process to be fermented">
            {volume.value} {volume.unit}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="Boil Volume">
          <Tooltip
            title="Target amount of liquid to be collected through lautering and sparging
to then be boiled"
          >
            {boil_volume.value} {boil_volume.unit}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="ABV">
          <Tooltip title="Alcohol by Volume">{abv}</Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="Target FG">
          <Tooltip title="Final Gravity; Measurement of relative density of the beer at end of fermentation">
            {target_fg}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="Target OG">
          <Tooltip title="Original Gravity; Measurement of the relative density of the wort before fermentation">
            {target_og}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="EBC">
          <Tooltip title="Measurement of colour used by european brewing convention">
            {ebc}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="SRM">
          <Tooltip title="â€“ Measurement of colour used by american society of brewing chemists">
            {srm}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="pH">
          <Tooltip title="Power of Hydrogen; Measurement of acidity or alkalinity">
            {ph}
          </Tooltip>
        </Descriptions.Item>
        <Descriptions.Item label="Attenuation level">
          <Tooltip title="Measurement of the percentage of sugars converted to alcohol and CO2">
            {attenuation_level}
          </Tooltip>
        </Descriptions.Item>
      </Descriptions>
      <Descriptions title="Food Pairings">
        {food_pairing.map((v,i) => (
          <Descriptions.Item key={i}>{v}</Descriptions.Item>
        ))}
      </Descriptions>
      <Descriptions title="Brewer's Tips"></Descriptions>
      <span>{brewers_tips}</span>
      <Drawer
        title="Ingredients"
        width={400}
        onClose={() => setShowIngredients(false)}
        visible={showIngredients}
      >
        <Title level={3}>Malt</Title>
        <Table
          pagination={false}
          dataSource={malt.map(({ amount: { value, unit }, ...rest }) => ({
            amount: `${value} ${unit}`,
            ...rest,
          }))}
          columns={[
            {
              title: "Name",
              dataIndex: "name",
            },
            {
              title: "Amount",
              dataIndex: "amount",
            },
          ]}
        />
        <Title level={3}>Hops</Title>
        <Table
          pagination={false}
          dataSource={hops.map(({ amount: { value, unit }, ...rest }) => ({
            amount: `${value} ${unit}`,
            ...rest,
          }))}
          columns={[
            {
              title: "Name",
              dataIndex: "name",
            },
            {
              title: "Amount",
              dataIndex: "amount",
            },
            {
              title: "Add",
              dataIndex: "add",
            },
            {
              title: "Attribute",
              dataIndex: "attribute",
            },
          ]}
        />
        <Title level={3}>Yeast</Title>
        {yeast}
      </Drawer>
      <Drawer
        title="Method"
        width={400}
        onClose={() => setShowMethod(false)}
        visible={showMethod}
      >
        <Title level={3}>Mash temp</Title>
        <Table
          pagination={false}
          dataSource={mash_temp.map(({ temp: { value, unit }, ...rest }) => ({
            temp: `${value} ${unit}`,
            ...rest,
          }))}
          columns={[
            {
              title: "Temperature",
              dataIndex: "temp",
            },
            {
              title: "Duration",
              dataIndex: "duration",
            },
          ]}
        />
        <Title level={3}>Fermentation</Title>
        <Descriptions>
          <Descriptions.Item label="Temperature">
            {fermentation.temp.value} {fermentation.temp.unit}
          </Descriptions.Item>
        </Descriptions>
        {!!twist && (
          <>
            <Title level={3}>Twist</Title>
            {twist.split(",").map((v, i) => (
              <div key={i}>{v}</div>
            ))}
          </>
        )}
      </Drawer>
    </Drawer>
  );
}
export default BeerDrawer;
