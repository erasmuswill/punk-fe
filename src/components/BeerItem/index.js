import ReadMoreReact from "read-more-react";
import Skeleton from "react-loading-skeleton";
import "./BeerItem.scss";

function BeerItem({
  abv,
  name,
  description,
  image_url,
  id,
  setModalId = () => {},
  loading,
}) {
  if (
    !loading &&
    (!abv || !name || !description || !image_url || !id || !setModalId)
  )
    return null;
  return (
    <div className={"beer-item"} key={id} onClick={() => setModalId(id)}>
      <div className="image">
        {loading ? (
          <Skeleton height={100} width={50} />
        ) : (
          <img
            height="100"
            src={loading ? "https://images.punkapi.com/v2/keg.png" : image_url}
            alt={`Label for ${name}`}
          />
        )}
      </div>
      <div className="name">
        <h3>{loading ? <Skeleton /> : name}</h3>
      </div>
      <span className="abv">{loading ? <Skeleton /> : `${abv}% alcohol`}</span>
      <div className="description">
        {loading ? (
          <Skeleton count={2} />
        ) : (
          <ReadMoreReact text={description} min={300} ideal={310} max={320} />
        )}
      </div>
    </div>
  );
}

export default BeerItem;
