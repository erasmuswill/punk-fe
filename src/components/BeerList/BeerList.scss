div.beer-list {
  padding: 16px;

  div.beer-item {
    margin-bottom: 12px;
    display: grid;
    grid-template-columns: 40px 1fr;
    grid-template-rows: 25px 27px auto;
    min-height: 144px;
    border: 1px solid transparent;

    &:hover {
      border: 1px solid #1890ff !important;
    }

    div.image {
      grid-row-start: 1;
      grid-row-end: 4;
      min-width: 50px;
      img {
        margin: 8px auto;
        display: block;
      }
    }

    div.name {
      margin: 0px 24px;
      h3 {
        color: #1890ff;
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
      }
    }

    div.description {
      margin: 0px 24px;
      grid-column-start: 2;
      grid-column-end: 3;
    }

    span.abv {
      color: #a9a9a9;
      margin: 4px 24px;
    }
  }
}

.beer-list-filters {
  margin: 8px;
  padding: 8px 16px;

  > div > div {
    box-shadow: none;
  }

  button[class*="rootPrimaryVariant"]:not(:disabled) {
    background-color: #1890ff;
  }
}
