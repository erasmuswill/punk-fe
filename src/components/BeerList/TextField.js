import React, { useState } from "react";
import { v4 as uuidv4 } from "uuid";
import { createUseStyles } from "react-jss";
import { useTheme } from "react-criteria";
import { DatePicker, Input } from "antd";
import moment from 'moment';

const useStyles = createUseStyles({
  rootGutterBottom: {
    marginBottom: "1em",
  },

  field: (theme) => ({
    width: "100%",
    padding: "1em",
    fontSize: "1em",
    color: "inherit",
    borderRadius: "2px",
    backgroundColor: "transparent",
    border: `solid 1px ${theme.container.borderColor}`,
  }),

  label: {
    display: "block",
    marginBottom: ".5em",
  },
});

function Textfield(props) {
  const {
    value,
    label,
    type = "text",
    gutterBottom = false,
    onChange: onChangeProp,
    ...otherProps
  } = props;

  const classes = useStyles(useTheme());
  const id = React.useMemo(() => uuidv4(), []);
  const [selDate,setDate]=useState(null);

  const onChange = React.useCallback(
    (ev) => {
      if (typeof onChangeProp !== "function") return;
      if (type === "date") {onChangeProp(ev.format("MM-YYYY")); setDate(ev)}
      else onChangeProp(ev.target.value.replace(/[^a-zA-Z0-9-\s]/,''));
    },
    [onChangeProp]
  );

  const rootElementClassName = React.useMemo(() => {
    const classNames = [];

    if (gutterBottom === true) classNames.push(classes.rootGutterBottom);

    return classNames.join(" ");
  }, [classes, gutterBottom]);

  return (
    <div className={rootElementClassName}>
      {label != null && (
        <label htmlFor={id} className={classes.label}>
          {label}
        </label>
      )}

      {type !== "date" ? (
        <Input
          id={id}
          type={type}
          value={value}
          onChange={onChange}
          className={classes.field}
          {...otherProps}
        />
      ) : (
        <DatePicker
          id={id}
          type={type}
          value={selDate}
          onChange={onChange}
          className={classes.field}
          {...otherProps}
        />
      )}
    </div>
  );
}

export default React.memo(Textfield);
