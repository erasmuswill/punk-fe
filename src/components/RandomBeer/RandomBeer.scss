div.random-beer {
  min-height: 212px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: center;
  align-items: center;
  border-radius: 12px;
  border: 1px solid #1890ff !important;
  padding: 16px;
  margin: 22px;

  div.title-block {
    flex-basis: 10%;
    img {
      margin: 8px auto;
      display: block;
    }
  }
  div.label-block {
    flex-basis: 60%;
    margin: 0px 24px;
  }
  div.cta-block {
    flex-basis: 20%;
    button {
      width: 150px;
    }
    button:first-of-type {
      margin-bottom: 16px;
    }
  }
}
