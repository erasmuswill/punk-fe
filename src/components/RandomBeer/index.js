import { Button, Typography } from "antd";
import { ReloadOutlined } from "@ant-design/icons";
import { useState, Suspense } from "react";
import { useRandomBeer } from "../../api";
import BeerDrawer from "../BeerDrawer";
import BeerItem from "../BeerItem";

const { Title } = Typography;

function RandomBeer() {
  const { beer = {}, loading, isValidating, mutate } = useRandomBeer();
  const [modalId, setModalId] = useState(null);
  return (
    <>
      <Title level={3}>I'm feeling lucky</Title>
      <BeerItem {...beer} setModalId={setModalId} />
      <Button
        type="primary"
        disabled={loading || isValidating}
        onClick={() => mutate()}
        icon={<ReloadOutlined spin={isValidating} />}
      >
        New random beer
      </Button>
      <BeerDrawer id={modalId} close={() => setModalId(null)} />
    </>
  );
}

const SuspensefulBeerDrawer = (props) => (
  <Suspense
    fallback={
      <>
        <Title level={3}>I'm feeling lucky</Title>
        <BeerItem loading />
        <Button loading disabled icon={<ReloadOutlined spin />}>
          New random beer
        </Button>
      </>
    }
  >
    <RandomBeer {...props} />
  </Suspense>
);

export default SuspensefulBeerDrawer;
