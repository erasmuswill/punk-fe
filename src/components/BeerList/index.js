import { useBeerList } from "../../api";
import InfiniteScroll from "react-infinite-scroll-component";
import Criteria from "react-criteria";
import ReadMoreReact from "read-more-react";
import "./BeerList.scss";
import { useState } from "react";
import Textfield from "./TextField";
import { Empty } from "antd";
import Title from "antd/lib/typography/Title";

function BeerList() {
  const [filter, setFilter] = useState([]);
  const {
    beers,
    isLoadingMore,
    size,
    setSize,
    hasNextPage,
  } = useBeerList(
    filter.map(({ type, value }) => ({
      [type]: value.replaceAll(" ", "_"),
    }))
  );
  const loadMoreRows = isLoadingMore ? () => {} : () => setSize(size + 1);
  return (
    <>
      <div className="beer-list-filters">
        <Title level={2}>Filters</Title>
        <Criteria
          data={filter}
          onChange={setFilter}
          criteria={{
            abv_gt: {
              label: "ABV Greater Than",
              addable: !filter.filter(({ type }) => type === "abv_gt").length,
              component: {
                component: Textfield,
                props: {
                  min: 0.1,
                  max: 100,
                  type: "number",
                  autoFocus: true,
                  placeholder: "Enter maximum ABV",
                },
              },
            },
            abv_lt: {
              label: "ABV Lower Than",
              addable: !filter.filter(({ type }) => type === "abv_lt").length,
              component: {
                component: Textfield,
                props: {
                  min: 0.1,
                  max: 100,
                  type: "number",
                  autoFocus: true,
                  placeholder: "Enter minimum ABV",
                },
              },
            },
            brewed_before: {
              label: "Brewed before",
              addable: !filter.filter(({ type }) => type === "brewed_before")
                .length,
              component: {
                component: Textfield,
                addable: false,
                props: {
                  type: "date",
                  picker: "month",
                  autoFocus: true,
                  placeholder: "Enter maximum first brew date",
                },
              },
            },
            beer_name: {
              label: "Beer Name",
              addable: !filter.filter(({ type }) => type === "beer_name").length,
              component: {
                component: Textfield,
                addable: false,
                props: {
                  autoFocus: true,
                  placeholder: "Enter a term to find beer",
                },
              },
            },
          }}
        />
      </div>
      <div className="beer-list">
        {beers && beers.length ? (
          <InfiniteScroll
            dataLength={beers.length}
            next={loadMoreRows}
            hasMore={hasNextPage}
            loader={<h4>Loading...</h4>}
            endMessage={
              beers.length > 20 && (
                <p style={{ textAlign: "center" }}>
                  <b>Yay! You have seen it all</b>
                </p>
              )
            }
          >
            {beers.map(({ abv, name, description, image_url }) => (
              <div className="beer-item">
                <div className="image">
                  <img height="100" src={image_url} alt={`Label for ${name}`} />
                </div>
                <div className="name">
                  <h3>{name}</h3>
                </div>
                <span className="abv">{abv}%</span>
                <div className="description">
                  <ReadMoreReact
                    text={description}
                    min={300}
                    ideal={310}
                    max={320}
                  />
                </div>
              </div>
            ))}
          </InfiniteScroll>
        ) : (
          <Empty />
        )}
      </div>
    </>
  );
}
export default BeerList;
