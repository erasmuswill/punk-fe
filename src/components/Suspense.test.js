import { render } from "@testing-library/react";
import React from "react";
import BeerDrawer from "./BeerDrawer";
import BeerItem from "./BeerItem";
import BeerList from "./BeerList";
import RandomBeer from "./RandomBeer";

jest.mock("react", () => {
  const React = jest.requireActual("react");
  React.Suspense = ({ fallback }) => fallback;
  return React;
});

describe("Suspense", () => {
  it("BeerItem", () => {
    const { asFragment } = render(<BeerItem loading />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("BeerList", () => {
    const { asFragment } = render(<BeerList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("RandomBeer", () => {
    const { asFragment } = render(<RandomBeer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("BeerDrawer", () => {
    const { asFragment, debug } = render(<BeerDrawer id={11} />);
    debug();
    expect(asFragment()).toMatchSnapshot();
  });
});
