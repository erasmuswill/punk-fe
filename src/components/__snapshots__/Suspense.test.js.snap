// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suspense BeerDrawer 1`] = `<DocumentFragment />`;

exports[`Suspense BeerItem 1`] = `
<DocumentFragment>
  <div
    class="beer-item"
  >
    <div
      class="image"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          style="width: 50px; height: 100px;"
        >
          ‌
        </span>
      </span>
    </div>
    <div
      class="name"
    >
      <h3>
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </h3>
    </div>
    <span
      class="abv"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
    </span>
    <div
      class="description"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Suspense BeerList 1`] = `
<DocumentFragment>
  <h3
    class="ant-typography"
  >
    All beer
  </h3>
  <div
    class="beer-list-filters loading"
  >
    <div
      class="ant-spin ant-spin-spinning"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
  <div
    class="beer-list"
  >
    <div
      class="beer-item"
    >
      <div
        class="image"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 50px; height: 100px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="name"
      >
        <h3>
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            >
              ‌
            </span>
          </span>
        </h3>
      </div>
      <span
        class="abv"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </span>
      <div
        class="description"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
    <div
      class="beer-item"
    >
      <div
        class="image"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 50px; height: 100px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="name"
      >
        <h3>
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            >
              ‌
            </span>
          </span>
        </h3>
      </div>
      <span
        class="abv"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </span>
      <div
        class="description"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
    <div
      class="beer-item"
    >
      <div
        class="image"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 50px; height: 100px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="name"
      >
        <h3>
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            >
              ‌
            </span>
          </span>
        </h3>
      </div>
      <span
        class="abv"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </span>
      <div
        class="description"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
    <div
      class="beer-item"
    >
      <div
        class="image"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 50px; height: 100px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="name"
      >
        <h3>
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            >
              ‌
            </span>
          </span>
        </h3>
      </div>
      <span
        class="abv"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </span>
      <div
        class="description"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
    <div
      class="beer-item"
    >
      <div
        class="image"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style="width: 50px; height: 100px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="name"
      >
        <h3>
          <span>
            <span
              class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            >
              ‌
            </span>
          </span>
        </h3>
      </div>
      <span
        class="abv"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </span>
      <div
        class="description"
      >
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Suspense RandomBeer 1`] = `
<DocumentFragment>
  <h3
    class="ant-typography"
  >
    I'm feeling lucky
  </h3>
  <div
    class="beer-item"
  >
    <div
      class="image"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          style="width: 50px; height: 100px;"
        >
          ‌
        </span>
      </span>
    </div>
    <div
      class="name"
    >
      <h3>
        <span>
          <span
            class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
          >
            ‌
          </span>
        </span>
      </h3>
    </div>
    <span
      class="abv"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
    </span>
    <div
      class="description"
    >
      <span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
        <span
          class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
        >
          ‌
        </span>
      </span>
    </div>
  </div>
  <button
    class="ant-btn ant-btn-loading"
    disabled=""
    type="button"
  >
    <span
      class="ant-btn-loading-icon"
    >
      <span
        aria-label="loading"
        class="anticon anticon-loading anticon-spin"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </span>
    <span>
      New random beer
    </span>
  </button>
</DocumentFragment>
`;
